src/conversion.rs

```diff
@@ -370,6 +372,60 @@ where
     }
 }
 
+impl<T: ToPyObject> ToPyObject for Box<T> {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        self.as_ref().to_object(py)
+    }
+}
+
+impl<T: Clone + IntoPy<PyObject>> IntoPy<PyObject> for Box<T> {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        self.as_ref().clone().into_py(py)
+    }
+}
+
+// impl<'a, T: FromPyObject<'a>> FromPyObject<'a> for Box<T> {
```

Comment:
I think the coherence error is correct insofar `Box<T>` is a "fundamental" type IIRC, i.e. downstream crates can `impl PyClass for Box<T>` with the same coherence rules as if the implemented `impl PyClass for T`.

I am not sure whether we can get around this without introducer said additional trait to decouple `FromPyObject`'s task to "produce a `T` given a `PyAny`" from the new (sealed) trait's "I consider this a transparent wrapper of its inner value get/set purposes", so maybe

```rust
trait PyPropertyWrapper<T>: Sealed {
  fn set(val: T) -> Self;
  fn get(&self) -> T;
}
```

?

Reply:
Thanks for letting me know about fundamental types, this explains the issue with `Box`!